src/components/MarketDetail/index.js

```diff
@@ -257,38 +275,46 @@ class MarketDetail extends Component {
   }
 
   renderControls() {
-    const { market, closeMarket } = this.props
+    const { market, closeMarket, defaultAccount } = this.props
     return (
       <div className="marketControls container">
         <div className="row">
-          {Object.keys(expandableViews).filter(view =>
-            typeof expandableViews[view].showCondition !== 'function' ||
-            expandableViews[view].showCondition(this.props),
-          ).map(view => (
-            <button
-              key={view}
-              type="button"
-              className={`
+          {Object.keys(expandableViews)
+            .filter(
+              view =>
+                typeof expandableViews[view].showCondition !== 'function' ||
+                expandableViews[view].showCondition(this.props),
+            )
+            .map(view => (
+              <button
+                key={view}
+                type="button"
+                className={`
                 marketControls__button
-                ${(view !== DEFAULT_VIEW && view === this.props.params.view)
-                || (view === DEFAULT_VIEW && view === this.props.params.view)
-                || (this.props.params.view === undefined && view === DEFAULT_VIEW) ? 'marketControls__button--active btn btn-primary'
-                : expandableViews[view].className}`
-              }
-              onClick={() => this.handleExpand(view)}
-            >
-              {expandableViews[view].label}
-            </button>
-          ))}
-          {market.stage !== MARKET_STAGES.MARKET_CLOSED ?
+                ${(view !== DEFAULT_VIEW && view === this.props.params.view) ||
+                (view === DEFAULT_VIEW && view === this.props.params.view) ||
+                (this.props.params.view === undefined && view === DEFAULT_VIEW)
+                  ? 'marketControls__button--active btn btn-primary'
+                  : expandableViews[view].className}`}
+                onClick={() => this.handleExpand(view)}
+              >
+                {expandableViews[view].label}
+              </button>
+            ))}
+          {market.stage !== MARKET_STAGES.MARKET_CLOSED &&
+          market.creator === defaultAccount &&
+          process.env.WHITELIST[defaultAccount] ? (
```

Comment:
`process.env.WHITELIST` is this not inside the config.json? Can that be used instead?

Reply:
We needed to have two kind of whitelists, one for production and one for development with the list of the default TestRPC accounts.
In the webpack config we check for the environment type (dev or prod) and pick productionWhitelist or developmentWhitelist:
https://github.com/gnosis/gnosis-management/blob/master/webpack.dev.config.js#L16

Reply:
Ah, nevermind. It's because of the dev/prod switch. Got it!